connector/oidc/oidc.go

```diff
@@ -284,12 +285,29 @@ func (c *oidcConnector) Refresh(ctx context.Context, s connector.Scopes, identit
 	return c.createIdentity(ctx, identity, token, refreshCaller)
 }
 
+func (c *oidcConnector) TokenIdentity(ctx context.Context, subjectTokenType, subjectToken string) (connector.Identity, error) {
+	var identity connector.Identity
+	token := &oauth2.Token{
+		AccessToken: subjectToken,
+	}
+	if subjectTokenType == "urn:ietf:params:oauth:token-type:access_token" {
+		token = token.WithExtra(map[string]any{
+			"id_token": subjectToken,
+		})
+	}
+	return c.createIdentity(ctx, identity, token, exchangeCaller)
+}
+
 func (c *oidcConnector) createIdentity(ctx context.Context, identity connector.Identity, token *oauth2.Token, caller caller) (connector.Identity, error) {
 	var claims map[string]interface{}
 
 	rawIDToken, ok := token.Extra("id_token").(string)
 	if ok {
-		idToken, err := c.verifier.Verify(ctx, rawIDToken)
+		verifier := c.verifier
+		if caller == exchangeCaller {
+			verifier = c.provider.Verifier(&oidc.Config{SkipClientIDCheck: true})
+		}
+		idToken, err := verifier.Verify(ctx, rawIDToken)
```

Comment:
I think this was a bug from me only testing with access tokens, testing with my new code, I get 

> time="2023-06-10T11:09:22Z" level=error msg="failed to verify subject token: oidc: error loading userinfo: oidc: user info endpoint is not supported by this provider"

Which I think just a restriction on the provider I used rather than a fundamental limitation of access -> id